src/printer.rs

```diff
@@ -37,26 +38,39 @@ impl<'a> Printer<'a> {
     }
 
     pub fn print_header(&mut self, filename: &str) -> Result<()> {
-        match self.options.style {
-            OptionsStyle::Full => {}
-            _ => return Ok(()),
+        if !self.options
+            .output_components
+            .contains(&OutputComponent::Header)
+        {
+            return Ok(());
         }
 
-        self.print_horizontal_line('┬')?;
-
-        writeln!(
-            self.handle,
-            "{}{} File {}",
-            " ".repeat(PANEL_WIDTH),
-            self.colors.grid.paint("│"),
-            self.colors.filename.paint(filename)
-        )?;
-
-        self.print_horizontal_line('┼')
+        if self.options
+            .output_components
+            .contains(&OutputComponent::Grid)
+        {
+            self.print_horizontal_line('┬')?;
+
+            writeln!(
+                self.handle,
+                "{}{} File {}",
+                " ".repeat(PANEL_WIDTH),
+                self.colors.grid.paint("│"),
+                self.colors.filename.paint(filename)
+            )?;
+
+            self.print_horizontal_line('┼')
+        } else {
+            writeln!(self.handle, "File {}", self.colors.filename.paint(filename))
+                .map_err(Into::into)
```

Comment:
Since I am returning the result here, no. It would complain about getting a `Result<_, io::Error>` instead of `Result<_, Error>`. I can use `?;` though and simply return `Ok(())` at the end instead.

Let me know if you think that would be easier to understand!